src/main/java/io/cdap/plugin/gcp/gcs/StorageClient.java

```diff
@@ -115,9 +115,67 @@ public void mapMetaDataForAllBlobs(String path, Consumer<Map<String, String>> fu
    * @param recursive whether to copy objects in all subdirectories
    * @param overwrite whether to overwrite existing objects
    * @throws IllegalArgumentException if overwrite is false and copying would overwrite an existing object
+   * @return copied blob object from google bucket storage or null if we are dealing with a directory/bucket
    */
-  public void copy(GCSPath sourcePath, GCSPath destPath, boolean recursive, boolean overwrite) {
+  public Blob copy(GCSPath sourcePath, GCSPath destPath, boolean recursive, boolean overwrite) {
+    //If sourcePath is not a bucket and sourcePath doesn't end with "/"
+    //For example: gs://plugin_376/csvexample.csv
+    //It means we are dealing with a single object which can be copied to the destination
+    if (!sourcePath.isBucket() && !sourcePath.getName().endsWith("/")) {
```

Comment:
Is there no other approach on how to check difference between file and directory?

Reply:
Well, apparently, for our case, this is the only way. 

If for example, we want to copy a _directory_ from source path to destination path and the source path is in this way --> **gs://cirus-test-bucket/output/csv**

We will have a **null** in return which means the directory cannot be found.
That's because google cloud storage can find directories only when we have a '/' character at the end of path, like in this way: **gs://cirus-test-bucket/output/csv/**

We've tried with Blob's _isDirectory()_ method but according to java doc: 
https://googleapis.dev/java/google-cloud-storage/latest/com/google/cloud/storage/BlobInfo.html#isDirectory--

**public boolean isDirectory()**
`Returns true if the current blob represents a directory. This can only happen if the blob is returned by Storage.list(String, Storage.BlobListOption...) when the Storage.BlobListOption.currentDirectory() option is used. When this is the case only getBlobId() and getSize() are set for the current blob: BlobId.getName() ends with the '/' character, BlobId.getGeneration() returns null and getSize() is 0.`

https://github.com/googleapis/java-storage/blob/4e5ee0398e700baf4f88224f66309e426f9532d7/google-cloud-storage/src/main/java/com/google/cloud/storage/BlobInfo.java#L931-L941

Reply:
I see. lgtm.